src/python/foglamp/configuration_manager.py

```diff
@@ -336,35 +364,73 @@
     """
     category_val_prepared = ''
     try:
-        # validate new category_val
+        # validate new category_val, set "value" from default
         category_val_prepared = await _validate_category_val(category_value, True)
         # check if category_name is already in storage
         category_val_storage = await _read_category_val(category_name)
         if category_val_storage is None:
-            return await _create_new_category(category_name, category_val_prepared, category_description)
+            await _create_new_category(category_name, category_val_prepared, category_description)
         else:
-            # validate category_val from storage
+            # validate category_val from storage, do not set "value" from default, reuse from storage value
             try:
                 category_val_storage = await _validate_category_val(category_val_storage, False)
+            # if validating category from storage fails, nothing to salvage from storage, use new completely
             except:
                 _logger.exception(
                     'category_value for category_name %s from storage is corrupted; using category_value without merge',
                     category_name)
+            # if validating category from storage succeeds, merge new and storage
             else:
                 category_val_prepared = await _merge_category_vals(category_val_prepared, category_val_storage, keep_original_items)
-            return await _update_category(category_name, category_val_prepared, category_description)
-
+                if json.dumps(category_val_prepared, sort_keys=True) == json.dumps(category_val_storage, sort_keys=True):
+                    return
+            await _update_category(category_name, category_val_prepared, category_description)
     except:
         _logger.exception(
             'Unable to create new category based on category_name %s and category_description %s and category_json_schema %s',
             category_name, category_description, category_val_prepared)
         raise
+    try:
+        _run_callbacks(category_name)
+    except:
+        _logger.exception(
+            'Unable to run callbacks for category_name %s', category_name)
+        raise
     return None
 
 
-def register_category(category_name, callback):
-    pass
+def register_interest(category_name, callback):
```

Comment:
can a module register interest of multiple category names? 
`register_interest(*categories, callback_module)` and call `register_interest("CAT1", "CAT2", callback_module="foglamp.blah")`

Reply:
I could implement it this way.. but it wasn't specified in the design doc. @MarkRiddoch what do you think?